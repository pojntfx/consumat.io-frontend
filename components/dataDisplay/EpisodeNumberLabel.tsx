import React from "react";
import { EpisodeNumber } from "../../types/episodeNumber";

type EpisodeNumberLabelProps = {
  episodeNumber: EpisodeNumber;
  className?: string;
};

function EpisodeNumberLabel({
  episodeNumber,
  className,
}: EpisodeNumberLabelProps) {
  return (
    <div className={"flex flex-row " + className}>
      <div className="font-medium">
        S
        {episodeNumber.season.toLocaleString("en", {
          minimumIntegerDigits: 2,
        })}
      </div>
      <div className="-mx-0.5 font-medium">ï¸±</div>
      <div className="font-medium">
        E
        {episodeNumber.episode.toLocaleString("en", {
          minimumIntegerDigits: 2,
        })}
      </div>
    </div>
  );
}

export default EpisodeNumberLabel;
