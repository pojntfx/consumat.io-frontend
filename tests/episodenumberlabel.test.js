import React from "react";
import renderer from "react-test-renderer";
import EpisodeNumberLabel from "../components/dataDisplay/EpisodeNumberLabel";

const exampleEpisodeNumber = {
  season: 2,
  episode: 5,
};
const exampleClassname = "ml-2";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <EpisodeNumberLabel
        className={exampleClassname}
        episodeNumber={exampleEpisodeNumber}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
