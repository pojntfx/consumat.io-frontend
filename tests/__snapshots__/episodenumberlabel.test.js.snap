// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-row ml-2"
>
  <div
    className="font-medium"
  >
    S
    02
  </div>
  <div
    className="-mx-0.5 font-medium"
  >
    ï¸±
  </div>
  <div
    className="font-medium"
  >
    E
    05
  </div>
</div>
`;
